mitodl/edx-platform
# Added legacy grade export functionality to the instructor dashboard

#### What are the relevant tickets?

Closes mitocw#74

#### What's this PR do?

Adds legacy grade export functionality back into the instructor dashboard (was removed after the dogwood releases)

#### Where should the reviewer start?

Probably `lms/templates/instructor/instructor_dashboard_2/grade_export.html` and `lms/static/js/instructor_dashboard/grade_export.js`.

#### How should this be manually tested?

- Set the `ENABLE_INSTRUCTOR_GRADE_EXPORT` feature to `True`
- Set a `REMOTE_GRADEBOOK_URL` feature value - you can ask me or Peter for the proper setting here
- Navigate to the `Export Grades` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-grade_export` - make sure you have staff/admin privileges)
- Click the buttons and make sure they work

#### Any background context you want to provide?

This grade exporting functionality last existed in the dogwood releases (eg: [dogwood.3-3](https://github.com/mitodl/edx-platform/tree/mitx/dogwood.3-3)) in something called the legacy instructor dashboard. As such there is a lot of functionality that was lifted out of the codebase as it existed then. I made several changes to match the patterns that currently exist in the instructor dashboard (the buttons make AJAX requests instead of submitting a general HTML form, etc.), but I tried to be careful not to overdo it with optimizing legacy code.

A few key files from the dogwood release that contained the legacy dashboard functionality:
- https://github.com/mitodl/edx-platform/blob/mitx/dogwood.3-3/lms/templates/courseware/legacy_instructor_dashboard.html
- https://github.com/mitodl/edx-platform/blob/mitx/dogwood.3-3/lms/djangoapps/instructor/views/legacy.py#L211
